trustfall_core/src/graphql_query/query.rs

```diff
@@ -125,49 +125,57 @@ fn try_get_query_root(document: &ExecutableDocument) -> Result<&Positioned<Field
 
     match &document.operations {
         DocumentOperations::Multiple(mult) => {
-            return Err(ParseError::MultipleOperationsInDocument(mult.values().next().unwrap().pos))
+            if mult.values().len() > 1 {
+                Err(ParseError::MultipleOperationsInDocument(
+                    mult.values()
+                        .nth(2)
+                        .expect("Could not iterate to second value in document.")
+                        .pos,
+                ))
+            } else if let Some(node) = mult.values().next() {
+                parse_operation_definition(node)
+            } else {
+                // This should be unreachable if someone is using the library correctly
+                panic!("These aren't the droids you're looking for.")
```

Comment:
I actually don't think its possible to hit this error with valid GraphQL LOL

Reply:
This is a fun message to write, but it'll be very frustrating if anyone actually hits it. I hate to be the fun police, but could we try to apply some empathy for the poor user that hits this case?
```suggestion
                unreachable!("Found a `DocumentOperations::Multiple()` with no query components. \
                              This shouldn't be possible, and is a bug. Please report it at \
                              https://github.com/obi1kenobi/trustfall/")
```